src/ray/gcs/gcs_server/store_client_kv.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/gcs/gcs_server/store_client_kv.h"
+
+#include <memory>
+
+namespace ray {
+namespace gcs {
+
+StoreClientInternalKV::StoreClientInternalKV(std::unique_ptr<StoreClient> store_client)
+    : delegate_(std::move(store_client)) {}
+
+void StoreClientInternalKV::Get(
+    const std::string &ns,
+    const std::string &key,
+    std::function<void(std::optional<std::string>)> callback) {
+  RAY_CHECK_OK(delegate_->AsyncGet(
+      ns, key, [callback = std::move(callback)](auto status, auto result) {
+        callback(result.has_value() ? std::optional<std::string>(result.value())
+                                    : std::optional<std::string>());
+      }));
+}
+
+void StoreClientInternalKV::Put(const std::string &ns,
+                                const std::string &key,
+                                const std::string &value,
+                                bool overwrite,
+                                std::function<void(bool)> callback) {
+  RAY_CHECK_OK(delegate_->AsyncPut(
+      ns, key, value, overwrite, [callback = std::move(callback)](bool result) {
```

Comment:
should we just put callback there?

  RAY_CHECK_OK(delegate_->AsyncPut(ns, key, value, overwrite, callback));